api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -67,6 +75,14 @@ public void setUserDAO(UserDAO dao) {
 		this.dao = dao;
 	}
 	
+	/**
+	 * @return the validTime for which the password reset activation key will be valid
+	 */
+	public int getValidTime() {
+		//if valid time is less that a minute or greater than 12hrs reset valid time to 1 minutes else set it to the required time.
+		return (validTime < MIN_VALID_TIME) || (validTime > MAX_VALID_TIME) ? DEFAULT_VALID_TIME : validTime;
```

Comment:
We said you should define a GP for this, that's why you have min, max and default values so that you can have some control on bad values that the admin sets